src/containers/pages/InterventionPlan/IRS/plan/index.tsx

```diff
@@ -1,79 +1,1419 @@
+
+          if (!planJurisdictionIdsToGet.length) {
+            return new Promise(resolve => resolve());
+          }
+
+          // build superset adhoc filter expression
+          let sqlFilterExpression = '';
+          for (let i = 0; i < planJurisdictionIdsToGet.length; i += 1) {
+            const jurId = planJurisdictionIdsToGet[i];
+            if (i) {
+              sqlFilterExpression += ' OR ';
+            }
+            sqlFilterExpression += `jurisdiction_id = '${jurId}'`;
+          }
+
+          // define superset params for
+          const planJurisdictionSupersetParams = superset.getFormData(1000, [
+            { sqlExpression: sqlFilterExpression },
+          ]);
+
+          return supersetService(SUPERSET_JURISDICTIONS_SLICE, planJurisdictionSupersetParams).then(
+            (jurisdictionResults: Jurisdiction[]) =>
+              fetchJurisdictionsActionCreator(jurisdictionResults)
+          );
+        }
+      );
+    }
+
+    // GET REMAINING JURISDICTIONS
+    if (!isFinalizedPlan) {
+      let sqlFilterExpression = '';
+      let idsToUse: string[] = [];
+      if (loadedJurisdictionIds.length || planJurisdictionIdsToGet.length) {
+        otherJurisdictionIdsToGet = allJurIds.filter((j: string) => {
+          return !loadedJurisdictionIds.includes(j) && !planJurisdictionIdsToGet.includes(j);
+        });
+
+        const otherJurisdictionIdsNotToGet = Array.from(
+          new Set([...planJurisdictionIdsToGet, ...loadedJurisdictionIds])
+        );
+
+        const doInclude = otherJurisdictionIdsToGet.length < otherJurisdictionIdsNotToGet.length;
+        idsToUse = doInclude ? otherJurisdictionIdsToGet : otherJurisdictionIdsNotToGet;
+
+        // build query params
+        for (let i = 0; i < idsToUse.length; i += 1) {
+          const jurId = idsToUse[i];
+          if (i) {
+            sqlFilterExpression += doInclude ? ' OR ' : ' AND ';
+          }
+          sqlFilterExpression += `id ${doInclude ? '=' : '!='} '${jurId}'`;
+        }
+      }
+
+      const otherJurisdictionSupersetParams = sqlFilterExpression.length
+        ? superset.getFormData(3000, [{ sqlExpression: sqlFilterExpression }])
+        : { row_limit: 3000 };
+
+      supersetService(SUPERSET_JURISDICTIONS_DATA_SLICE, otherJurisdictionSupersetParams).then(
+        (jurisdictionResults: FlexObject[] = []) => {
+          const jurisdictions = jurisdictionResults.map(j => {
+            const { id, parent_id, name, geographic_level } = j;
+            const jurisdiction: Jurisdiction = {
+              geographic_level: geographic_level || 0,
+              jurisdiction_id: id,
+              name: name || null,
+              parent_id: parent_id || null,
+            };
+            return jurisdiction;
+          });
+          return fetchJurisdictionsActionCreator(jurisdictions);
+        }
+      );
+    }
+  }
+
+  public componentWillReceiveProps(nextProps: IrsPlanProps) {
```

Comment:
I feel uncomfortable using `componentWillReceiveProps` given that it is deprecated 🙈 for a good reason IMO.  I think if we have to use it, we should have a corresponding issue to investigate the possibility of replacing it with something else.

Reply:
Oh interesting, not sure how to get around this off the top of my head, but if this is the case then we should probably have the issue be for updating the GisidaWrapper component as well.

Reply:
Yeah I'm good with this moving to a technical debt issue, for tackling later.

Reply:
Cool deal, issue created at #237